src/lamplib/src/genny/toolchain.py

```diff
@@ -39,11 +39,10 @@ def _create_compile_environment(
 
     else:
         # For cmake and ctest
-        cmake_bin_relative_dir = {
-            "linux": "downloads/tools/cmake-3.24.0-linux/cmake-3.24.0-linux-x86_64/bin",
-            "osx": "downloads/tools/cmake-3.24.0-osx/cmake-3.24.0-macos-universal/CMake.app/Contents/bin",
-        }[triplet_os]
-        paths.insert(0, os.path.join(toolchain_dir, cmake_bin_relative_dir))
+        if triplet_os == "linux":
+            # Only use bundled cmake on Linux because bundled cmake doesn't work on OS X due to integrity checks.
+            # It's easier just to ask users to "brew install cmake"
+            paths.insert(0, os.path.join(toolchain_dir, "downloads/tools/cmake-3.24.0-linux/cmake-3.24.0-linux-x86_64/bin"))
```

Comment:
Specifically, it should be an "if linux -> check architecture".

Reply:
(The OS X binary was universal, which is why this wasn't broken for that before)

Reply:
There isn't really a clean way to refactor this, since right now for some reason ARM64 linux uses the installed ninja, but x64 linux does not, and OS X doesn't as well, but OS X also doesn't use cmake as of this change. The combination means that any minimal conditional will be messy.